klippy/extras/heater_generic.py

```diff
@@ -0,0 +1,37 @@
+# Support for a generic heater
+#
+# Copyright (C) 2019  John Jardine <john@gprime.net>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+
+import logging
+
+class PrinterHeaterGeneric:
+    def __init__(self, config, heater_name=None):
+        self.printer = config.get_printer()
+        gcode_id = config.get("gcode_id")
+        self.heater = self.printer.lookup_object('heater').setup_heater(config, gcode_id, heater_name)
+        self.gcode = self.printer.lookup_object("gcode")
+        self.gcode.register_command("SET_HEATER_TEMPERATURE", None)
+        self.gcode.register_command("SET_HEATER_TEMPERATURE", self.cmd_SET_HEATER_TEMPERATURE, desc=self.cmd_SET_HEATER_TEMPERATURE_help)
```

Comment:
I would prefer SET_HEATER_TEMPERATURE to take the name of the config section instead of the g-code id.  (G-code is horrible - I'd prefer to avoid its idiosyncrasies in new commands.)  So, this should use the gcode.register_mux_command() tool with the heater_name.

Separately, I think it may be preferable to move the SET_HEATER_TEMPERATURE to heater.py so it could be used by all heaters.

Reply:
Ah, that makes sense. What do you envision the parameters should be? SET_HEATER_TEMPERATURE NAME=<heater_name> TARGET=<target_temp> ? Any other suggestions?

Reply:
Actually now I'm leaning more towards "HEATER" as the parameter name.